src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -155,15 +199,37 @@ public IAIState pickUpMaterial()
 
         if (currentStructure.getStage() != StructureIterator.Stage.DECORATE)
         {
+            //Log.getLogger().info("Check if item is decoration and now is not decoration time");
             needsCurrently = needsCurrently.and(stack -> !ItemStackUtils.isDecoration(stack));
+            //Log.getLogger().info( needsCurrently.toString());
+
         }
 
         if (InventoryUtils.hasItemInItemHandler(worker.getInventoryCitizen(), needsCurrently))
         {
-            return getState();
+            // here still needs to be work done. Here the implementation follows to calculate all items
+            // which equals the needed item in the inventory of the worker follows.
+            // at the moment we dump the inventory after each step anyway. so it is not needed.
+            // the change enables that the builder can pickup more than only one stack.
+            // This part is unfinished and may be removed entierly later...
+            // get the amount in Inventory and compares it to the needed amount.
+//            int amount = needsCurrently.and(stack -> ItemStack.getSize(stack));
+//
+//            int otheramount = 0;
+//            int slot = InventoryUtils.findFirstSlotInItemHandlerNotEmptyWith(worker.getInventoryCitizen(), needsCurrently);
+//            if (slot != -1) {
+//                ItemStack myStack = worker.getInventoryCitizen().getStackInSlot(slot);
+//                otheramount = myStack.getCount();
+//            }
+//            Log.getLogger().info("But has "+ otheramount +" already in local inventory ");
+            //return getState();
+            return GATHERING_REQUIRED_MATERIALS;
```

Comment:
I don't understand this change. If the worker has the item in the inventory, then it doesn't have to pick more up, ain't I right?

Reply:
(You can also remove all the commented out code here)

Reply:
done